### PR TITLE
[Security] improve some firewall config comments

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #19398
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: src/Symfony/Bundle/SecurityBundle/Security/FirewallConfig.php [hunk 1]
```diff
@@ -28,7 +28,20 @@ class FirewallConfig
...
+     * @param string      $name
+     * @param string      $requestMatcher
+     * @param bool        $securityEnabled
+     * @param bool        $stateless
+     * @param string|null $provider
+     * @param string|null $context
+     * @param string|null $entryPoint
+     * @param string|null $accessDeniedHandler
+     * @param string|null $accessDeniedUrl
+     * @param string      $userChecker
```
Comment (javiereguiluz) [inline src/Symfony/Bundle/SecurityBundle/Security/FirewallConfig.php]: Isn't it a bit strange to have this mandatory argument after all those optional nullable arguments?

**Review by javiereguiluz — COMMENTED** [review]
👍

File: src/Symfony/Bundle/SecurityBundle/Security/FirewallConfig.php (hunk 1 above)
Comment (xabbuh) [inline src/Symfony/Bundle/SecurityBundle/Security/FirewallConfig.php]: Yes indeed. We could still reorder the arguments.

**Comment by chalasr** [comment]
The SecurityDataCollectorTest needs to be updated due to the argument becoming mandatory. If you want, I can do this change and btw reorder the arguments, so you just correct the doc here.

File: src/Symfony/Bundle/SecurityBundle/Security/FirewallConfig.php [hunk 2]
```diff
@@ -104,20 +117,23 @@ public function getUserChecker()
...
-     * @return string The access_denied_handler service id
+     * @return string|null The access_denied_handler service id if configured, null otherwise
      */
     public function getAccessDeniedHandler()
     {
         return $this->accessDeniedHandler;
     }
 
+    /**
+     * @return string|null The access_denied_handler service id if configured, null otherwise
```
Comment (stof) [inline src/Symfony/Bundle/SecurityBundle/Security/FirewallConfig.php]: wrong. This is not a service id

File: src/Symfony/Bundle/SecurityBundle/Security/FirewallConfig.php (hunk 1 above)
Comment (stof) [inline src/Symfony/Bundle/SecurityBundle/Security/FirewallConfig.php]: we should indeed reorder them before being stuck with BC issues

**Comment by xabbuh** [comment]
@chalasr I reverted the change to not allowing `null` anymore for the user checker argument. Do you wanna create an additional PR with that change?

**Comment by xabbuh** [comment]
@chalasr I reverted the change to not allowing `null` anymore for the user checker argument. Do you wanna create an additional PR with that change?

**Comment by chalasr** [comment]
@xabbuh see #20407

**Comment by chalasr** [comment]
@xabbuh see #20407

**Comment by chalasr** [comment]
@xabbuh see #20407

**Comment by chalasr** [comment]
@xabbuh see #20407

**Comment by chalasr** [comment]
@xabbuh see #20407

**Comment by chalasr** [comment]
@xabbuh see #20407

**Comment by chalasr** [comment]
@xabbuh see #20407

**Comment by chalasr** [comment]
@xabbuh see #20407

**Comment by chalasr** [comment]
@xabbuh see #20407

**Comment by chalasr** [comment]
@xabbuh see #20407

**Comment by chalasr** [comment]
@xabbuh see #20407

**Comment by chalasr** [comment]
@xabbuh see #20407

**Comment by chalasr** [comment]
@xabbuh see #20407

**Comment by chalasr** [comment]
@xabbuh see #20407

**Comment by chalasr** [comment]
@xabbuh see #20407

**Comment by chalasr** [comment]
@xabbuh see #20407

**Comment by fabpot** [comment]
Thank you @xabbuh.

**Pull request merged**